### ISSUE TITLE
derive EthAbiType would cause clippy::unwrap-used lint error

### ISSUE THREAD

**Issue by zsluedem** [gakonst/ethers-rs#2053]
**Version**
`ethers = { git = "https://github.com/gakonst/ethers-rs.git", rev = "3ed83d5dd38e7819605511631aaddcc3e472427e", features = ["ethers-solc"]  }`

**Platform**
Linux will-MS-7B89 5.15.0-58-generic #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

**Description**

I tried this code:

```rust
  #[derive(Clone, EthAbiType)]
  pub struct UserOperation {
      pub sender: Address,
  }
```

cargo clippy --  -D clippy::unwrap_used

Lint error would show
```
error: used `unwrap()` on `an Option` value
    |
  |         pub sender: Address,
    |         ^^^
    |
    = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_used
    = note: requested on the command line with `-D clippy::unwrap-used`

error: could not compile `aa-bundler` due to previous error

```

**Comment by mattsse** [comment]
It'd be great if you could attach the expanded code

**Comment by zsluedem** [comment]
@mattsse I created a pr #2056  for it

```rust 
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use ethers::prelude::EthAbiType;
use ethers::types::{Address, Bytes};
pub struct UserOperation {
    pub sender: Address,
    pub paymaster_and_data: Bytes,
    pub signature: Bytes,
}
#[automatically_derived]
impl ::core::clone::Clone for UserOperation {
    #[inline]
    fn clone(&self) -> UserOperation {
        UserOperation {
            sender: ::core::clone::Clone::clone(&self.sender),
            paymaster_and_data: ::core::clone::Clone::clone(&self.paymaster_and_data),
            signature: ::core::clone::Clone::clone(&self.signature),
        }
    }
}
impl ::ethers::core::abi::AbiType for UserOperation {
    fn param_type() -> ::ethers::core::abi::ParamType {
        ::ethers::core::abi::ParamType::Tuple(<[_]>::into_vec(
            #[rustc_box]
            ::alloc::boxed::Box::new([
                <Address as ::ethers::core::abi::AbiType>::param_type(),
                <Bytes as ::ethers::core::abi::AbiType>::param_type(),
                <Bytes as ::ethers::core::abi::AbiType>::param_type(),
            ]),
        ))
    }
}
impl ::ethers::core::abi::AbiArrayType for UserOperation {}
impl ::ethers::core::abi::Tokenizable for UserOperation
where
    Address: ::ethers::core::abi::Tokenize,
    Bytes: ::ethers::core::abi::Tokenize,
    Bytes: ::ethers::core::abi::Tokenize,
{
    fn from_token(
        token: ::ethers::core::abi::Token,
    ) -> ::std::result::Result<Self, ::ethers::core::abi::InvalidOutputType>
    where
        Self: Sized,
    {
        if let ::ethers::core::abi::Token::Tuple(tokens) = token {
            if tokens.len() != 3usize {
                return Err(::ethers::core::abi::InvalidOutputType({
                    let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                        &["Expected ", " tokens, got ", ": "],
                        &[
                            ::core::fmt::ArgumentV1::new_display(&3usize),
                            ::core::fmt::ArgumentV1::new_display(&tokens.len()),
                            ::core::fmt::ArgumentV1::new_debug(&tokens),
                        ],
                    ));
                    res
                }));
            }
            let mut iter = tokens.into_iter();
            Ok(Self {
                sender: ::ethers::core::abi::Tokenizable::from_token(iter.next().unwrap())?,
                paymaster_and_data: ::ethers::core::abi::Tokenizable::from_token(
                    iter.next().unwrap(),
                )?,
                signature: ::ethers::core::abi::Tokenizable::from_token(iter.next().unwrap())?,
            })
        } else {
            Err(::ethers::core::abi::InvalidOutputType({
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["Expected Tuple, got "],
                    &[::core::fmt::ArgumentV1::new_debug(&token)],
                ));
                res
            }))
        }
    }
    fn into_token(self) -> ::ethers::core::abi::Token {
        ::ethers::core::abi::Token::Tuple(<[_]>::into_vec(
            #[rustc_box]
            ::alloc::boxed::Box::new([
                self.sender.into_token(),
                self.paymaster_and_data.into_token(),
                self.signature.into_token(),
            ]),
        ))
    }
}
impl ::ethers::core::abi::TokenizableItem for UserOperation
where
    Address: ::ethers::core::abi::Tokenize,
    Bytes: ::ethers::core::abi::Tokenize,
    Bytes: ::ethers::core::abi::Tokenize,
{
}
fn main() {
    let _a = UserOperation {
        sender: Address::default(),
        paymaster_and_data: Bytes::default(),
        signature: Bytes::default(),
    };
}

```

**Comment by zsluedem** [comment]
fixed in #2056
